doit/action.py

```diff
@@ -390,6 +390,9 @@ def execute(self, out=None, err=None):
         elif isinstance(returned_value, dict):
             self.values = returned_value
             self.result = returned_value
+        elif isinstance(returned_value, TaskFailed) \
```

Comment:
The second argument of `isinstance()` accepts a tuple of types. https://docs.python.org/3/library/functions.html#isinstance